mos-platform/cpm65/stack.S

```diff
@@ -0,0 +1,7 @@
+.global __stack
```

Comment:
Is there any way to have the stack start at the end of memory and grow downward?

Note that heap created by malloc assumes the usual C layout: text/data, then heap growing up, followed by free space, followed by stack growing down. This will cap both the stack and heap size to 2K.

Reply:
Technically yes, but I'd rather not? The issue here is that the command processor lives at the top of memory, allowing it to be overwritten by programs which need the extra 2kB of RAM (and, for CP/M-65, the zero page it consumes). Putting the stack high commits us to either always overwriting the command processor or else never overwriting it. I'll also note that the heap is actually above the stack in my model.

What I was planning (not done yet) was to use linker magic so that if a program refers to the heap, then the command processor gets overwritten and a warm-boot command gets inserted into the shutdown code --- this tells the BDOS to reload the command processor. Otherwise, the program just returns normally.

Reply:
Ah, makes sense.